### PR TITLE
Futurize stage 1 with Black format

### PR DESCRIPTION
**Pull request by mmphego**
This PR includes 3 stages namely:
- Clean up #105 
- Futurize stage 1 #106 
- Black formatting. #107 

The reason for creating this PR is due to cascaded the PR's that were based on LIFO, as illustrated below. It is cumbersome to merge these PR's to master without merging #106 

```
Master
  |
  --> https://github.com/ska-sa/tango-simlib/pull/105
	  |
	  --> https://github.com/ska-sa/tango-simlib/pull/106
		  |
		  --> https://github.com/ska-sa/tango-simlib/pull/106
			  |
			  --> https://github.com/ska-sa/tango-simlib/pull/106
```

### CODE REVIEW

**Comment by mmphego** [comment]
Squash and merge this PR upon approval and close #107

File: tango_simlib/utilities/base_parser.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
```
Comment (ajoubertza) [inline tango_simlib/utilities/base_parser.py]: These imports still aren't in the right place, but it doesn't seem like that is going to be fixed.

Comment (mmphego) [reply tango_simlib/utilities/base_parser.py]: They are fixed in #108 
See https://github.com/ska-sa/tango-simlib/pull/108/files#diff-19

**Comment by mmphego** [comment]
> Very hard to review this - which is why the plan was to do small PRs instead (and those have already been reviewed, so now we are duplicating work).

I know, but we couldn't merge PR's #105, #106 and #107 without merging #108 as they are all interdependent, hence why after PR's #105, #106 and #107 where approved I then created this PR which merges all of them into a single PR.

Not sure if all this makes sense. :thinking:

**Pull request merged**